# HG changeset patch
# User John Donoghue
# Date 1426855464 14400
#      Fri Mar 20 08:44:24 2015 -0400
# Node ID 28a9e59d07149e2546a73a7daac66179d405961a
# Parent  f7846f0ea6db0b16284e644baa1143cb13aec413
cross-tools: Add ability to build cross built octave-config

* configure.ac: rename --enable-cross-mkoctfile to --enable-cross-tools

* src/Makefile.am: rename AMCOND_CROSS_MKOCTFILE to AMCOND_CROSS_TOOLS,
  add rule to build $(host_triplet)-octave-config$(BUILD_EXEEXT)

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -2952,30 +2952,30 @@ fi
 AM_CONDITIONAL([AMCOND_BUILD_GUI], [test $build_gui = yes])
 AM_CONDITIONAL([AMCOND_HAVE_QSCINTILLA],
                [test "$octave_cv_lib_qscintilla" = yes])
 AM_CONDITIONAL([WIN32_TERMINAL], [test $win32_terminal = yes])
 AC_SUBST(QT_CPPFLAGS)
 AC_SUBST(QT_LDFLAGS)
 AC_SUBST(QT_LIBS)
 
-# do we want to create cross-mkoctfile ?
-cross_mkoctfile=no
-AC_ARG_ENABLE([cross-mkoctfile],
-  [AS_HELP_STRING([--enable-cross-mkoctfile],
-    [build cross-mkoctfile if cross compiling])],
-  [if test "$enableval" = yes; then cross_mkoctfile=yes; fi])
-
-if test "$cross_mkoctfile" = yes; then
+# do we want to create cross-mkoctfile etc ?
+cross_tools=no
+AC_ARG_ENABLE([cross-tools],
+  [AS_HELP_STRING([--enable-cross-tools],
+    [build cross tools (mkoctfile, octave-config) if cross compiling])],
+  [if test "$enableval" = yes; then cross_tools=yes; fi])
+
+if test "$cross_tools" = yes; then
   if test "$cross_compiling" = no; then
-    AC_MSG_WARN([ignoring --enable-cross-mkoctfile when not cross compiling])
-    cross_mkoctfile=no
+    AC_MSG_WARN([ignoring --enable-cross-tools when not cross compiling])
+    cross_tools=no
   fi
 fi
-AM_CONDITIONAL([AMCOND_CROSS_MKOCTFILE], [test $cross_mkoctfile = yes])
+AM_CONDITIONAL([AMCOND_CROSS_TOOLS], [test $cross_tools = yes])
 
 
 ### Decide whether or not to install build logs with Octave.
 
 install_build_logs=no
 AC_ARG_ENABLE([install-build-logs],
   [AS_HELP_STRING([--enable-install-build-logs],
     [install build logs (e.g. config.log) with Octave])],
@@ -3136,17 +3136,17 @@ Octave is now configured for $canonical_
   Build Java interface:               $build_java
   Do internal array bounds checking:  $BOUNDS_CHECKING
   Build static libraries:             $STATIC_LIBS
   Build shared libraries:             $SHARED_LIBS
   Dynamic Linking:                    $ENABLE_DYNAMIC_LINKING $DL_API_MSG
   Include support for GNU readline:   $USE_READLINE
   64-bit array dims and indexing:     $USE_64_BIT_IDX_T
   OpenMP SMP multithreading:          $USE_OPENMP
-  Build cross mkoctfile:              $cross_mkoctfile
+  Build cross tools:                  $cross_tools
 ])
 
 warn_msg_printed=false
 
 OCTAVE_CONFIGURE_WARNING_SUMMARY
 
 if test $ENABLE_DYNAMIC_LINKING = yes; then
   if test $SHARED_LIBS = no; then
diff --git a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -126,27 +126,35 @@ mkoctfile_LDADD = $(top_builddir)/libgnu
 
 octave_config_SOURCES =
 nodist_octave_config_SOURCES = octave-config.cc
 octave_config_LDADD = \
   $(top_builddir)/libinterp/corefcn/libcorefcn.la \
   $(top_builddir)/libgnu/libgnu.la \
   $(LIBS)
 
-# if building cross octfile
 OCTAVE_CROSS_TOOLS=
-if AMCOND_CROSS_MKOCTFILE
+if AMCOND_CROSS_TOOLS
+# building cross mkoctfile
 OCTAVE_CROSS_TOOLS += $(host_triplet)-mkoctfile$(BUILD_EXEEXT)
 $(host_triplet)-mkoctfile$(BUILD_EXEEXT): $(host_triplet)-mkoctfile.cc
 	$(BUILD_CXX) -o $(host_triplet)-mkoctfile$(BUILD_EXEEXT) -Dgnulib='' -Doctave_idx_type=int $(DEFAULT_INCLUDES) $(BUILD_CXXFLAGS) $(BUILD_LDFLAGS) $(host_triplet)-mkoctfile.cc
-mostlyclean-local:
-	-rm -f $(TARGET)-mkoctfile$(BUILD_EXEEXT)
-
 $(host_triplet)-mkoctfile.cc: mkoctfile.in.cc Makefile
 	$(AM_V_GEN)$(do_subst_cross_config_vals)
+
+# building cross octave-config
+OCTAVE_CROSS_TOOLS += $(host_triplet)-octave-config$(BUILD_EXEEXT)
+$(host_triplet)-octave-config$(BUILD_EXEEXT): $(host_triplet)-octave-config.cc
+	$(BUILD_CXX) -o $(host_triplet)-octave-config$(BUILD_EXEEXT) -Dgnulib='' -Doctave_idx_type=int $(DEFAULT_INCLUDES) $(BUILD_CXXFLAGS) $(BUILD_LDFLAGS) $(host_triplet)-octave-config.cc
+$(host_triplet)-octave-config.cc: octave-config.in.cc Makefile
+	$(AM_V_GEN)$(do_subst_default_vals)
+
+mostlyclean-local:
+	-rm -f $(OCTAVE_CROSS_TOOLS)
+
 endif
 
 all-local: $(OCTAVE_VERSION_LINKS) $(OCTAVE_CROSS_TOOLS)
 
 octave-config.cc: octave-config.in.cc Makefile
 	$(AM_V_GEN)$(do_subst_default_vals)
 
 mkoctfile.cc: mkoctfile.in.cc Makefile
